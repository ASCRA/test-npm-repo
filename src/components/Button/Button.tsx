import React from 'react';
import styled, { css } from 'styled-components';
// import { LoadingSpinner } from '../Spinners/LoadingSpinner';

export interface ContainerProps {
  rounded?: boolean;
  outlined?: boolean;
  subtle?: boolean;
  disabled?: boolean;
  busy?: boolean;
}

export interface ButtonProps extends ContainerProps {
  style?: React.CSSProperties;
  children?: React.ReactChild;
}

const Container = styled.div<ButtonProps>`
  position: relative;
  width: 100%;
  height: 55px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #007aff;
  border-radius: 10px;
  font-weight: 500;
  font-size: 17px;
  line-height: 16px;
  text-align: center;
  letter-spacing: -0.14px;
  color: #ffffff;
  transition: transform 0.075s ease;
  user-select: none;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;

  &:active {
    transform: scale(0.95, 0.95);
  }

  &:hover {
    background-color: #0072ef;
  }

  ${({ rounded }) =>
    rounded &&
    css`
      border-radius: 100px;
    `}

  ${({ outlined }) =>
    outlined &&
    css`
      background-color: #ffffff;
      color: #007aff;
      border: 1px solid #e5e5e5;

      &:hover {
        background-color: #ffffff;
        border-color: #bababa;
      }

      &:active {
        border-color: #bababa;
      }
    `}

  ${({ subtle }) =>
    subtle &&
    css`
      background-color: #f2f2f2;
      color: #007aff;

      &:hover {
        background-color: #f9f9f9;
      }
    `}

  ${({ disabled }) =>
    disabled &&
    css`
      opacity: 0.5;
      pointer-events: none;
    `}

  ${({ busy }) =>
    busy &&
    css`
      pointer-events: none;
    `}
`;

const Button = (props: ButtonProps) => {
  const { rounded, outlined, subtle, disabled, busy, children, style } = props;

  return (
    <Container {...props}>
      {children}
      {/* {busy ? <LoadingSpinner color='inherit' /> : <>{children}</>} */}
    </Container>
  );
};

export default Button;
