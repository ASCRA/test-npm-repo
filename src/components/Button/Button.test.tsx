import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import Button from './Button';

test('it works', () => {
  const tree = renderer.create(<Button subtle />).toJSON()
  expect(tree).toMatchSnapshot()
})