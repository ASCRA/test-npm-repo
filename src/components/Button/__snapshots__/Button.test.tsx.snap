// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  position: relative;
  width: 100%;
  height: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #007aff;
  border-radius: 10px;
  font-weight: 500;
  font-size: 17px;
  line-height: 16px;
  text-align: center;
  -webkit-letter-spacing: -0.14px;
  -moz-letter-spacing: -0.14px;
  -ms-letter-spacing: -0.14px;
  letter-spacing: -0.14px;
  color: #ffffff;
  -webkit-transition: -webkit-transform 0.075s ease;
  -webkit-transition: transform 0.075s ease;
  transition: transform 0.075s ease;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  background-color: #f2f2f2;
  color: #007aff;
}

.c0:active {
  -webkit-transform: scale(0.95,0.95);
  -ms-transform: scale(0.95,0.95);
  transform: scale(0.95,0.95);
}

.c0:hover {
  background-color: #0072ef;
}

.c0:hover {
  background-color: #f9f9f9;
}

<div
  className="c0"
/>
`;
